Library/Homebrew/tap.rb

```diff
@@ -362,6 +362,29 @@ def link_completions_and_manpages
     end
   end
 
+  def fix_remote_configuration
+    return unless remote.include? "github.com"
+
+    current_upstream_head = Utils.popen_read("git", "-C", path, "symbolic-ref", "refs/remotes/origin/HEAD")
+                                 .sub(%r{^refs/remotes/origin/}, "").chomp
+    Utils.popen_read("git", "-C", path, "ls-remote", "--heads", "--exit-code", "origin", current_upstream_head)
+    return if $CHILD_STATUS.success?
+
+    ohai "#{name}: Unable to find the remote branch (#{current_upstream_head})", <<~EOS
+      Searching for a new origin/HEAD...
```

Comment:
```suggestion
    ohai "#{name}: Unable to find the remote branch (#{current_upstream_head})"
    puts "Searching GitHub API for a new origin/HEAD..."
```

probably want to ensure the `Homebrew::EnvConfig.no_github_api?` case is handled here.

Reply:
Should this state the tap name it's acting upon?

Reply:
> probably want to ensure the `Homebrew::EnvConfig.no_github_api?` case is handled here.

Good call

> Should this state the tap name it's acting upon?

This is what the output of a failed `brew update` followed by `brew tap --repair` looks like (before making any suggested changes):

```console
$ brew update
fatal: couldn't find remote ref refs/heads/trunk
Error: Fetching /usr/local/Homebrew/Library/Taps/rylan12/homebrew-branch-name-test failed!
==> Some taps failed to update!
The following taps failed to update:
  rylan12/branch-name-test
Try running `brew tap --repair`

$ brew tap --repair
==> rylan12/branch-name-test: Unable to find the remote branch (trunk)
Searching for a new origin/HEAD...
Changed branch name from trunk to main
```

So the `Searching for...` and `Changed branch name...` messages appear under the `ohai` header which does include the tap name. If there are multiple taps that need repairing, each will be listed under its own heading. I'm thinking about modifying the output to look something like this:

```console
$ brew tap --repair
==> rylan12/branch-name-test
Unable to find the remote branch (trunk). Searching for a new origin/HEAD...
Changed branch name from trunk to main
==> user/repo
Unable to find the remote branch (master). Searching for a new origin/HEAD...
Changed branch name from master to main
```

I'd also like to remove the double messaging from `brew update` which might go hand-in-hand with detecting the specific error type.

Reply:
> So the `Searching for...` and `Changed branch name...` messages appear under the `ohai` header which does include the tap name

🆒 👍🏻 

> I'd also like to remove the double messaging from `brew update` which might go hand-in-hand with detecting the specific error type.

👍🏻